src/main/resources/tpcds_2_4/q23b.sql

```diff
@@ -0,0 +1,50 @@
+--q23b.sql--
+
+
+ with frequent_ss_items as
+ (select substr(i_item_desc,1,30) itemdesc,i_item_sk item_sk,d_date solddate,count(*) cnt
+  from store_sales, date_dim, item
+  where ss_sold_date_sk = d_date_sk
+    and ss_item_sk = i_item_sk
+    and d_year in (2000, 2000+1, 2000+2,2000+3)
+  group by substr(i_item_desc,1,30),i_item_sk,d_date
+  having count(*) > 4),
+ max_store_sales as
+ (select max(csales) tpcds_cmax
+  from (select c_customer_sk,sum(ss_quantity*ss_sales_price) csales
+        from store_sales, customer, date_dim
+        where ss_customer_sk = c_customer_sk
+         and ss_sold_date_sk = d_date_sk
+         and d_year in (2000, 2000+1, 2000+2,2000+3)
+        group by c_customer_sk) x),
+ best_ss_customer as
+ (select c_customer_sk,sum(ss_quantity*ss_sales_price) ssales
+  from store_sales
+      ,customer
+  where ss_customer_sk = c_customer_sk
+  group by c_customer_sk
+  having sum(ss_quantity*ss_sales_price) > (50/100.0) *
```

Comment:
The mismatch between the substitution variables and the English question is a red herring.  There are answer sets for qualification queries so it's easy to validate what is correct.

I come back to understanding the goal here.  There really should be just two forms of queries: 

- The 1GB qualification version with substitution variables from the spec.  Presto uses these in their tests ([link](https://github.com/prestodb/presto/tree/master/presto-benchto-benchmarks/src/main/resources/sql/presto/tpcds))
- Query templates that are used to generate query files for a given data scale factor.  Different substitution variables are used for difference data sizes so it's technically incorrect to use the templates to generate query files for 1GB but then use those against 10TB.